lib/find-vs2017.js

```diff
@@ -0,0 +1,47 @@
+var log = require('npmlog')
+  , execSync = require('child_process').execSync
+  , path = require('path')
+
+var hasCache = false
+  , cache = null
+
+function findVS2017() {
+  if (hasCache)
+    return cache
+
+  hasCache = true
+
+  var ps = 'PowerShell -ExecutionPolicy Unrestricted -Command '
+  var csFile = path.join(__dirname, 'Find-VS2017.cs')
+  var psQuery = ps + '"&{Add-Type -Path \'' + csFile + '\'; [VisualStudioConfiguration.Main]::Query()}" 2>&1'
+
+  var vsSetup
+  try {
+    var vsSetupRaw = execSync(psQuery, { encoding: 'utf8' })
```

Comment:
I kept the sync version from the original PR because making it async would have required deeper changes in `build.js` and `configure.js`. This is no longer true after the change to how MSBuild is found, so I'll change to async.

I made sure this works when node-gyp is installed to a location with spaces in its path.

Powershell is always in the path, unless is has been tampered with. The location of the binary seems to be stable, I'll change this to use it directly to avoid issues.